docs/pages/code/operations/distinct.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace App;
+
+use Closure;
+use loophp\collection\Collection;
+
+include __DIR__ . '/../../../../vendor/autoload.php';
+
+// Example 1 -> Using the default callback, with scalar values
+$collection = Collection::fromIterable(['a', 'b', 'a', 'c'])
+    ->distinct(); // [0 => 'a', 1 => 'b', 3 => 'c']
+
+// Example 2 -> Using one custom callback, with object values
+final class User
+{
+    private string $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    public function name(): string
+    {
+        return $this->name;
+    }
+}
+
+$users = [
+    new User('foo'),
+    new User('bar'),
+    new User('foo'),
+    new User('a'),
+];
+
+$collection = Collection::fromIterable($users)
+    ->distinct(
+        static fn (User $left): Closure => static fn (User $right): bool => $left->name() === $right->name()
+    ); // [0 => User<foo>, 1 => User<bar>, 3 => User<a>]
```

Comment:
can you add another example which just uses the accessor callback? I assume you'd do:

`->distinct(null, static fn (User $user) => $user->name())`

Reply:
also would be good to have all cases in the spec:

- no callback
- only comparator callback
- only accessor callback
- both comparator and accessor callbacks

Reply:
Ok ! I will continue to work on this in the evening.

Reply:
- [x] no callback (it was there already)
- [x] only comparator callback (it was there already)
- [x] only accessor callback (just added in https://github.com/loophp/collection/pull/111/commits/875e2ae28c97dc3fc14a5b989436041266e6d250)
- [x] both comparator and accessor callbacks (it was there already)